File: Java/94.java
__________________________________________________________________________________________________
sample 0 ms submission
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        final List<Integer> ret = new ArrayList<>();
        inorderTraversal(root, ret);
        return ret;
    }
    
    private void inorderTraversal(TreeNode root, List<Integer> ret) {
        if (root == null) return;
        inorderTraversal(root.left, ret);
        ret.add(root.val);
        inorderTraversal(root.right, ret);
    }
}
__________________________________________________________________________________________________
sample 33424 kb submission
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new LinkedList<>();
        Stack<TreeNode> stack = new Stack<>();
        TreeNode cur = root;
        while(cur!=null || !stack.isEmpty()){
            while(cur!=null){
                stack.push(cur);
                cur = cur.left;
            }
            cur =stack.pop();
            res.add(cur.val);
            cur = cur.right;
        }
        return res;
    }
}
__________________________________________________________________________________________________

File: Python3/94.py
__________________________________________________________________________________________________
sample 20 ms submission
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, node: TreeNode) -> List[int]:
        run = True 
        stack = []
        out = []
        currentNode = node
        while run:
            if currentNode:
                stack.append(currentNode)
                currentNode = currentNode.left

            else:
                if stack:
                    currentNode = stack.pop()
                    out.append(currentNode.val)
                    currentNode = currentNode.right

                else:
                    run = False
        return out
__________________________________________________________________________________________________
sample 12956 kb submission
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        traversal = []
        self.getInOrder(root, traversal)
        return traversal
        
    def getInOrder(self, root, traversal):
        if not root:
            return
        self.getInOrder(root.left, traversal)
        traversal.append(root.val)
        self.getInOrder(root.right, traversal)
__________________________________________________________________________________________________
